client/ayon_core/pipeline/publish/lib.py

```diff
@@ -460,8 +460,19 @@ class method 'def apply_settings(cls, project_settings)'
             )
             apply_plugin_settings_automatically(plugin, plugin_settins, log)
 
+        # Pyblish already operated a filter based on host.
+        # But applying settings might have changed "hosts"
+        # value in plugin so re-filter.
+        plugin_hosts = getattr(plugin, "hosts", None)
+        if (
+            plugin_hosts
+            and "*" not in plugin_hosts
+            and host_name not in plugin_hosts
+        ):
```

Comment:
Requires import `from pyblish.plugin import host_is_compatible`
```suggestion
        if not host_is_compatible(plugin):
```

Reply:
Yep, thanks a lot both for your feedback, I'll adjust this PR accordingly 🙏